common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,258 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory;
 
+import android.os.Build;
+import android.util.Base64;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.exception.ErrorStrings;
+import com.microsoft.identity.common.internal.logging.Logger;
 import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.PKCEChallenge;
 
+import java.io.UnsupportedEncodingException;
 import java.net.URL;
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
 
 public class AzureActiveDirectoryAuthorizationRequest extends AuthorizationRequest {
+    protected static final String ENCODING_UTF8 = "UTF_8";
+    private static final String TAG = StringExtensions.class.getSimpleName();
+    private static final String RESOURCE = "resource";
+    private static final String LOGIN_HINT = "login_hint";
+    private static final String ADAL_ID_PLATFORM = "x-client-SKU";
```

Comment:
I bet we can share these values... 😃

Reply:
Yes, it shared in aad and mssts but it is not a standard request para for oauth2. As Shane mentioned before we should put all the constants only in its calling class, I have moved these constants from AuthenticationConstants to its calling classes.